pkg/execution/executor/executor.go

```diff
@@ -1103,8 +1103,13 @@ func (e *executor) HandleResponse(ctx context.Context, i *runInstance, resp *sta
 	if len(resp.Generator) > 0 {
 		// Handle generator responses then return.
 		if serr := e.HandleGeneratorResponse(ctx, i, resp); serr != nil {
+			var compileError *expressions.CompileError
+
 			// If this is an error compiling async expressions, fail the function.
-			if strings.Contains(serr.Error(), "error compiling expression") {
+			if errors.As(serr, &compileError) {
+				resp.Output = nil
+				resp.Err = compileError.Message()
```

Comment:
This might not be the smoothest way to surface the error. If you have any other ideas, please let me know!

Reply:
combined with the custom error this seems entirely reasonable